fendermint/app/src/app.rs

```diff
@@ -291,6 +299,37 @@ where
         let state = self.committed_state()?;
         Ok((state.state_params, state.block_height))
     }
+
+    fn verify_ipc_message(&self, ipc_message: IPCMessage) -> bool {
+        match ipc_message {
+            IPCMessage::TopDown(finality) => self.parent_finality.check_finality(&finality),
+            IPCMessage::BottomUp => unimplemented!(),
+        }
+    }
+
+    /// Verifies the list of transactions to see if they are valid. Returns true if all the txns
+    /// are valid and false if any one of the txns is invalid.
+    fn verify_messages(&self, txns: &Vec<Bytes>) -> bool {
+        for tx in txns {
+            let is_ok = match serde_json::from_slice::<ChainMessage>(tx) {
```

Comment:
If you take a look at the `BytesMessageInterpreter` you will see that we use IPLD binary format for messages, not JSON. The Cosmos SDK uses JSON, which may be why you thought Tendermint does as well, but it's actually agnostic, and I thought we should stick to the Filecoin convention of IPLD (not that I don't think JSON wouldn't be more convenient). 

So this check should not be here, or exist in this format, because checking this is exactly the purpose of `BytesMessageInterpreter`. I added this check in https://github.com/consensus-shipyard/fendermint/pull/185